services/workflows-service/src/data-migration/scripts/sync.ts

```diff
@@ -0,0 +1,307 @@
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from '@/app.module';
+import { DataSyncTables, PrismaClient } from '@prisma/client';
+import * as path from 'path';
+import { WorkflowDefinitionRepository } from '@/workflow-defintion/workflow-definition.repository';
+import console from 'console';
+import { DATA_MIGRATION_FOLDER_RELATIVE_PATH } from './consts';
+import { MD5 as objectMd5 } from 'object-hash';
+import deepDiff from 'deep-diff';
+import stableStringify from 'json-stable-stringify';
+import { AppLoggerService } from '@/common/app-logger/app-logger.service';
+import { InputJsonValue, NullableJsonNullValueInput } from '@/types';
+import { env } from '@/env';
+
+type Envionment = 'sandbox' | 'production' | 'development' | 'local';
+
+export type SyncedObject = {
+  crossEnvKey: string;
+  tableName: string;
+  columns: {
+    [key: string]: any;
+  };
+  syncConfig: {
+    strategy: 'replace' | 'partial-deep-merge' | 'upsert';
+  };
+  syncedEnvironments: string[];
+  dryRunEnvironments: string[];
+};
+
+// @TODO: map to repositories after adding transaction support - its important since we have addtional validation there
+const tableNamesMap = {
+  WorkflowDefinition: 'workflowDefinition',
+};
```

Comment:
Will improve TS inference
```suggestion
} as const;
```